src/VirtualClient/VirtualClient.Actions/PostgreSQL/PostgreSQLServerExecutor.cs

```diff
@@ -354,6 +368,210 @@ private async Task CreateDatabaseAsync(EventContext telemetryContext, Cancellati
+
+                    switch (distroInfo.LinuxDistribution)
+                    {
+                        case LinuxDistribution.Ubuntu:
+                        case LinuxDistribution.Debian:
+                            workingDirectory = this.Combine(this.PostgreSqlPackagePath, "ubuntu");
+                            break;
+                    }
+
+                    MemoryInfo memoryInfo = await this.SystemManagement.GetMemoryInfoAsync(cancellationToken);
+                    long totalMemoryKiloBytes = memoryInfo.TotalMemory;
+                    int bufferSizeInMegaBytes = Convert.ToInt32(totalMemoryKiloBytes * 0.75 / 1024);
+
+                    using (IProcessProxy process = await this.ExecuteCommandAsync(
+                        this.Combine(workingDirectory, inMemoryScript),
+                        $"{bufferSizeInMegaBytes}",
+                        workingDirectory,
+                        telemetryContext,
+                        cancellationToken,
+                        runElevated: true))
+                    {
+                        if (!cancellationToken.IsCancellationRequested)
+                        {
+                            await this.LogProcessDetailsAsync(process, telemetryContext);
+                            process.ThrowIfWorkloadFailed();
+                        }
+                    }
+                }
+            }
+        }
+
+        private async Task PrepareBalancedScenarioAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                string diskPathsArgument = string.Empty;
+                string diskFilter = "osdisk:false";
+
+                IEnumerable<Disk> disks = await this.SystemManagement.DiskManager.GetDisksAsync(cancellationToken)
+                        .ConfigureAwait(false);
+
+                if (disks?.Any() != true)
+                {
+                    throw new WorkloadException(
+                        "Unexpected scenario. The disks defined for the system could not be properly enumerated.",
+                        ErrorReason.WorkloadUnexpectedAnomaly);
+                }
+
+                IEnumerable<Disk> disksToTest = DiskFilters.FilterDisks(disks, diskFilter, this.Platform).ToList();
+
+                if (disksToTest?.Any() != true)
+                {
+                    throw new WorkloadException(
+                        "Expected disks to test not found. Given the parameters defined for the profile action/step or those passed " +
+                        "in on the command line, the requisite disks do not exist on the system or could not be identified based on the properties " +
+                        "of the existing disks.",
+                        ErrorReason.DependencyNotFound);
+                }
+
+                if (await this.CreateMountPointsAsync(disksToTest, cancellationToken).ConfigureAwait(false))
```

Comment:
Let's discuss more about this, ideally we should not let individual executors handle this if it's a common disk operation